src/app/course/challenge/list-of-teams/list-of-teams.component.ts

```diff
@@ -61,4 +61,11 @@ export class ListOfTeamsComponent implements OnInit {
     isInvited(team: Team): boolean {
         return team.who_can_join.includes(this.course.course_reg.id)
     }
+
+    isJoinAble(team: Team, event: CourseEvent): boolean {
+        return (team.is_private? this.isInvited(team) : !this.isInvited(team) )
+            && !this.isInTeam(team)
+            && !event.is_not_available_yet
```

Comment:
I still think there are some problems with this.
For one, if a team is not private we are checking `!this.isInvited(team)` what if the team was open and he was invited? Why are we not allowing them to join?